src/plugins/discover/public/application/main/discover_main_app.tsx

```diff
@@ -93,6 +93,8 @@ export function DiscoverMainApp(props: DiscoverMainProps) {
     resetSavedSearch(savedSearch.id);
   }, [resetSavedSearch, savedSearch]);
 
+  useSavedSearchAliasMatchRedirect({ savedSearch, spaces, history });
```

Comment:
Why do we need to pass history instance from services instead of `useHistory` one?

Reply:
it's not a history instance, it's a function that passed in here:

https://github.com/elastic/kibana/blob/3730dd0779ed8efd74aee88f57422781ec1ac122/src/plugins/discover/public/kibana_services.ts#L36

I do agree we can simplify this, but I'd rather do this in a separate stuff, since I only intended to move out the redirect code of our layout code :)